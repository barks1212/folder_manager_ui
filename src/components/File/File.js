import React from 'react';
import PropTypes from 'prop-types';

import { folderDataType } from '../../types/types';

const file = ({ contents }) => (
    <div>
        <ul>
            <li>{contents.type}</li>
            <li>{contents.name}</li>
            <li>{contents.added}</li>

        </ul>
    </div>
);

file.propTypes = {
    contents: PropTypes.shape(folderDataType)
};

export default file;