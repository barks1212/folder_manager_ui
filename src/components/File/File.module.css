.File {
    background: yellow;
    width: 10rem;
    height: auto;
    border: 2px dotted black;
    display: flex;
    justify-content: center;
    margin: 5px;
}

.File ul {
    list-style-type: none;
    margin: 0;
    padding: 20px;
    text-align: center;
}

.Name {
    font-weight: bold;
}