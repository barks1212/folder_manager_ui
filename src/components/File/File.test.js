import React from 'react';
import { render, cleanup } from '@testing-library/react'

import File from './File';

afterEach(cleanup);

describe('file', () => {
    const mockContents = {
        type: 'pdf',
        name: 'test',
        added: '2017-01-06'
    }

    it('should render', () => {
        const {asFragment} = render(<File contents={mockContents}/>)
        expect(asFragment()).toMatchSnapshot();
    })
})