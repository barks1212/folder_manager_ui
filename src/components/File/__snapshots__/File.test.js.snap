// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`file should render 1`] = `
<DocumentFragment>
  <div>
    <ul>
      <li>
        pdf
      </li>
      <li>
        test
      </li>
      <li>
        2017-01-06
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
